routing/turns_generator.hpp

```diff
@@ -122,7 +140,13 @@ TurnDirection GetRoundaboutDirection(bool isIngoingEdgeRoundabout, bool isOutgoi
  * \param turnInfo is used for cashing some information while turn calculation.
  * \param turn is used for keeping the result of turn calculation.
  */
-void GetTurnDirection(Index const & index, turns::TurnInfo & turnInfo, TurnItem & turn);
+void GetTurnDirection(Index const & index, RoutingMapping & mapping, turns::TurnInfo & turnInfo,
+                      TurnItem & turn);
 
+/*!
+ * \brief Finds UTurn started from current segment and returns how many segments it lasts.
+ * Returns 0 otherwise.
+ */
+size_t CheckUTurnOnRoute(vector<LoadedPathSegment> const & segments, size_t currentSegment, TurnItem & turn);
```

Comment:
It's quite non-obvious that curentSegment must be greater than zero. Is it possible to preserve zero-based indexing? Please, specify restrictions on currentSegment in the comment for the function.